compiler/rustc_ast_lowering/src/asm.rs

```diff
@@ -27,11 +28,13 @@ impl<'a, 'hir> LoweringContext<'a, 'hir> {
                 .emit();
         }
 
-        let mut clobber_abi = None;
+        let mut clobber_abis = FxHashSet::default();
```

Comment:
Use a `HashMap` keyed on the ABI, otherwise 2 identical ABIs with different spans will both be included.